raccoongang/edx-platform
# [Baobab] add sso middleware

**Description:** If user is not authenticated on EDX side, redirect him to SSO provider.
If user is already authenticated with SSO, EDX authenticates user and redirects him to the dashboard.

**Youtrack:** https://youtrack.raccoongang.com/agiles/104-148/105-166?tab=chart&issue=BAOBAB-144

**Configuration instructions:** 
Add `ForceSSOAuthenticationMiddleware` to the `ADDITIONAL_MIDDLEWARE_CLASSES` env settings

**Testing instructions:**

1. Go to SSO portal and authenticate. Go to edx lms, user should be authenticated and redirected to the dashboard
2. Go to edx lms, if user is not authenticated, he should be redirected to SSO. Authorize user, check he's redirected to the dashboard
